core/src/main/java/jenkins/plugins/DetachedPluginsManager.java

```diff
@@ -0,0 +1,214 @@
+
+
+    /**
+     * Record of which plugins which removed from core and when.
+     */
+    private static final List<DetachedPlugin> DETACHED_LIST;
+    /**
+     * Implicit dependencies that are known to be unnecessary and which must be cut out to prevent a dependency cycle among bundled plugins.
+     */
+    private static final Set<String> BREAK_CYCLES;
+
+    static {
+        try (InputStream is = ClassicPluginStrategy.class.getResourceAsStream("/jenkins/split-plugins.txt")) {
+            DETACHED_LIST = ImmutableList.copyOf(configLines(is).map(line -> {
+                String[] pieces = line.split(" ");
+                return new DetachedPluginsManager.DetachedPlugin(pieces[0], pieces[1] + ".*", pieces[2]);
+            }).collect(Collectors.toList()));
+        } catch (IOException x) {
+            throw new ExceptionInInitializerError(x);
+        }
+        try (InputStream is = ClassicPluginStrategy.class.getResourceAsStream("/jenkins/split-plugin-cycles.txt")) {
+            BREAK_CYCLES = ImmutableSet.copyOf(configLines(is).collect(Collectors.toSet()));
+        } catch (IOException x) {
+            throw new ExceptionInInitializerError(x);
+        }
+    }
+
+    private DetachedPluginsManager() {
+    }
+
+    /**
+     * Returns all the plugin dependencies that are implicit based on a particular Jenkins version
+     *
+     * @since 2.0
+     */
+    @Nonnull
+    public static List<PluginWrapper.Dependency> getImpliedDependencies(String pluginName, String jenkinsVersion) {
+        List<PluginWrapper.Dependency> out = new ArrayList<>();
+        for (DetachedPlugin detached : DETACHED_LIST) {
+            // don't fix the dependency for itself, or else we'll have a cycle
+            if (detached.shortName.equals(pluginName)) {
+                continue;
+            }
+            if (BREAK_CYCLES.contains(pluginName + ' ' + detached.shortName)) {
+                LOGGER.log(Level.FINE, "skipping implicit dependency {0} → {1}", new Object[]{pluginName, detached.shortName});
+                continue;
+            }
+            // some earlier versions of maven-hpi-plugin apparently puts "null" as a literal in Hudson-Version. watch out for them.
+            if (jenkinsVersion == null || jenkinsVersion.equals("null") || new VersionNumber(jenkinsVersion).compareTo(detached.splitWhen) <= 0) {
+                out.add(new PluginWrapper.Dependency(detached.shortName + ':' + detached.requiredVersion));
+                LOGGER.log(Level.FINE, "adding implicit dependency {0} → {1} because of {2}",
+                           new Object[]{pluginName, detached.shortName, jenkinsVersion});
+            }
+        }
+        return out;
+    }
+
+    /**
+     * Get the list of all plugins that have ever been {@link DetachedPlugin detached} from Jenkins core.
+     *
+     * @return A {@link List} of {@link DetachedPlugin}s.
+     */
+    @Restricted(NoExternalUse.class)
```

Comment:
is it needed as all the class is restricted? (the same with some other methods)

Reply:
`Restricted` should generally be the default behavior when adding code. And yes, that is intended because I don't wish to make it a public API forever :-).

Reply:
I meant, the whole class is already Restricted. Do you need to annotate the method too? Isn't it enough with the annotation on the class?

Reply:
done, removed from methods. I left it also on the internal class, just in case the access-modifier handles this incorrectly. Should be harmful in any case.